tools/validators/instance_validator/validate/telemetry_validator.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+
+TRANSLATION = "translation"
+STATES = "states"
+UNITS = "unit_values"
+
+class TelemetryValidator(object):
+  """TODO"""
+
+  def __init__(self, entities, timeout, callback):
+    super().__init__()
+    self.entities = entities
+    self.timeout = timeout
+    self.callback = callback
+    self.validated_entities = {}
+    self.validation_errors = {}
+
+  def StartTimer(self, timeout):
+    threading.Timer(timeout, lambda: self.callback(self)).start()
+
+  def AllEntitiesValidated(self):
+    return self.entities.len() == self.validated_entities.len()
+
+  def CheckAllEntitiesValidated(self):
```

Comment:
this could be more declaratively named. Ex: CallbackIfComplete()

Reply:
Done